applications/GeoMechanicsApplication/tests/test_submerged_construction_of_excavation.py

```diff
@@ -0,0 +1,205 @@
+        self.unit_weight_of_water = 10.0e3  # N/m3
+        self.model_width = 65.0  # m
+        self.excavation_width = 15.0  # m
+        self.unsaturated_clay_layer_thickness = 2.0  # m
+        self.saturated_clay_layer_thickness = 18.0  # m
+        self.saturated_sand_layer_thickness = 30.0  # m
+        self.excavated_middle_clay_layer_thickness = 8.0  # m
+        self.excavated_lower_clay_layer_thickness = 10.0  # m
+        self.weight_of_diaphragm_wall = 10.0e3  # N / m
+        self.length_of_diaphragm_wall = 30.0  # m
+        self.distributed_surface_load = 5.0e3  # N / m / m
+        self.load_edge_length = 5.0  # m
+
+    def total_reaction_y_from_output_data(self, output_data, time, node_ids):
+        reactions = GiDOutputFileReader.nodal_values_at_time(
+            "REACTION", time, output_data, node_ids=node_ids
+        )
+        return sum([reaction[1] for reaction in reactions])
+
+    def calculate_weight_of_all_soil(self):
+        return self.model_width * (
+                self.unsaturated_clay_layer_thickness * self.unsaturated_unit_weight_of_clay
+                + self.saturated_clay_layer_thickness * self.saturated_unit_weight_of_clay
+                + self.saturated_sand_layer_thickness * self.saturated_unit_weight_of_sand
+        )
+
+    def calculate_weight_of_excavated_clay_upper_right(self):
+        return (
+                self.unsaturated_clay_layer_thickness
+                * self.excavation_width
+                * self.unsaturated_unit_weight_of_clay
+        )
+
+    def calculate_weight_of_excavated_clay_middle_right(self):
+        return (
+                self.excavated_middle_clay_layer_thickness
+                * self.excavation_width
+                * self.saturated_unit_weight_of_clay
+        )
+
+    def calculate_weight_of_excavated_clay_lower_right(self):
+        return (
+                self.excavated_lower_clay_layer_thickness
+                * self.excavation_width
+                * self.saturated_unit_weight_of_clay
+        )
+
+    def calculate_weight_of_water_after_second_excavation(self):
+        return (
+                self.excavated_middle_clay_layer_thickness
+                * self.excavation_width
+                * self.unit_weight_of_water
+        )
+
+    def calculate_weight_of_water_after_third_excavation(self):
+        return (
+                (
+                        self.excavated_middle_clay_layer_thickness
+                        + self.excavated_lower_clay_layer_thickness
+                )
+                * self.excavation_width
+                * self.unit_weight_of_water
+        )
+
+    def calculate_weight_of_diaphragm_wall(self):
+        return self.weight_of_diaphragm_wall * self.length_of_diaphragm_wall
+
+    def calculate_total_vertical_surface_load(self):
+        return self.distributed_surface_load * self.load_edge_length
+
+    def check_vertical_reaction(self, output_reader, project_path, stage, expected):
```

Comment:
Perhaps we can make one name slightly more descriptive (and remove a parameter that could be a local variable)?
```suggestion
    def check_vertical_reaction(self, project_path, stage, expected_total_vertical_reaction):
```

Reply:
done